src/mount.rs

```diff
@@ -0,0 +1,655 @@
+                Ok(Arc::new(index.read_hunk(hunk_id)?.unwrap_or_default()))
+            })
+            .cloned()
+    }
+
+    pub fn get_or_open_tree(&self, policy: BandSelectionPolicy) -> Result<Arc<StoredTree>> {
+        let band_id = self.archive.resolve_band_id(policy)?;
+        self.stored_tree_cache
+            .lock()
+            .unwrap()
+            .try_get_or_insert(band_id, || {
+                debug!("Opening band {}", band_id);
+
+                let stored_tree = self
+                    .archive
+                    .open_stored_tree(BandSelectionPolicy::Specified(band_id))?;
+
+                Ok(Arc::new(stored_tree))
+            })
+            .cloned()
+    }
+
+    pub fn get_or_create_hunk_helper(&self, stored_tree: &StoredTree) -> Result<Arc<HunkHelper>> {
+        let band_id = stored_tree.band().id();
+        self.hunk_helper_cache
+            .lock()
+            .unwrap()
+            .try_get_or_insert(band_id, || {
+                /* Inform the user that this band has been cached as this is most likely a heavy operaton (cpu and memory wise) */
+                info!("Caching files for band {}", stored_tree.band().id());
+
+                let helper = HunkHelper::from_index(&stored_tree.band().index())?;
+                Ok(Arc::new(helper))
+            })
+            .cloned()
+    }
+
+    fn parse_path_band_policy(
+        components: &mut dyn Iterator<Item = Cow<'_, str>>,
+    ) -> Option<BandSelectionPolicy> {
+        match components.next().as_deref() {
+            Some("latest") => Some(BandSelectionPolicy::Latest),
+            Some("all") => components
+                .next()
+                .and_then(|band_id| band_id.parse::<BandId>().ok())
+                .map(BandSelectionPolicy::Specified),
+            _ => None,
+        }
+    }
+
+    fn band_id_to_directory_info(&self, policy: BandSelectionPolicy) -> Option<DirectoryInfo> {
+        let stored_tree = self.get_or_open_tree(policy).ok()?;
+        let band_info = stored_tree.band().get_info().ok()?;
+
+        let timestamp = unix_time_to_windows(
+            band_info.start_time.unix_timestamp(),
+            band_info.start_time.unix_timestamp_nanos() as u32,
+        );
+
+        Some(DirectoryInfo {
+            directory_name: format!("{}", band_info.id),
+            directory_attributes: DIRECTORY_ATTRIBUTES,
+
+            creation_time: timestamp,
+            last_access_time: timestamp,
+            last_write_time: timestamp,
+        })
+    }
+
+    fn serve_dir(&self, path: &Path) -> Result<Vec<DirectoryEntry>> {
+        debug!("Serving directory {}", path.display());
+
+        let mut components = path
+            .components()
+            .map(Component::as_os_str)
+            .map(OsStr::to_string_lossy);
+
+        let target_band = match components.next().as_deref() {
+            None => {
+                /* Virtual root, display channel selection */
```

Comment:
By "channel" do you mean "band?"

Reply:
Yes, a relict of an older implementation.